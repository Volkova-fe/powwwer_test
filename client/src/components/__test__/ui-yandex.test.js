import React from 'react';
import TestRenderer from 'react-test-renderer';
import { render, screen, fireEvent } from '@testing-library/react';

import { Button, EmailInput, PasswordInput } from '@ya.praktikum/react-developer-burger-ui-components';

describe('Тестирование компонента Login', () => {

	it('Кнопка без текста рендерится без ошибок', () => {
		const button = TestRenderer
			.create(<Button />)
			.toJSON()
		expect(button).toMatchSnapshot()
	})

	it('Кнопка c типом secondary рендерится без ошибок', () => {
		const button = TestRenderer
			.create(<Button type="secondary"/>)
			.toJSON()
		expect(button).toMatchSnapshot()
	})

	it('Кнопка с htmlType="button" рендерится без ошибок', () => {
		const button = TestRenderer
			.create(<Button htmlType="button"/>)
			.toJSON()
		expect(button).toMatchSnapshot()
	})

	it('Кнопка с htmlType="submit" рендерится без ошибок', () => {
		const button = TestRenderer
			.create(<Button htmlType="submit"/>)
			.toJSON()
		expect(button).toMatchSnapshot()
	})

	it('Кнопка без текста рендерится без ошибок', () => {
		const button = TestRenderer
			.create(<Button />)
			.toJSON()
		expect(button).toMatchSnapshot()
	})

	it('Кнопка size="small" рендерится без ошибок', () => {
		const button = TestRenderer
			.create(<Button size="small"/>)
			.toJSON()
		expect(button).toMatchSnapshot()
	})

	it('Кнопка size="medium" рендерится без ошибок', () => {
		const button = TestRenderer
			.create(<Button size="medium"/>)
			.toJSON()
		expect(button).toMatchSnapshot()
	})

	it('Кнопка size="large" рендерится без ошибок', () => {
		const button = TestRenderer
			.create(<Button size="large"/>)
			.toJSON()
		expect(button).toMatchSnapshot()
	})

	it('Нажатие на кнопку вызывает колбек', () => {
		window.alert = jest.fn();

		render(<Button data-testId="button" onClick={() => { alert('Успешный вызов колбека') }} />)

		const button = screen.getByTestId("button");
		fireEvent.click(button);

		expect(window.alert).toHaveBeenCalledWith('Успешный вызов колбека');
	});

	it('EmailInput без текста рендерится без ошибок', () => {
		const emailInput = TestRenderer
			.create(<EmailInput />)
			.toJSON()
		expect(emailInput).toMatchSnapshot()
	})

	it('PasswordInput без текста рендерится без ошибок', () => {
		const passwordInput = TestRenderer
			.create(<PasswordInput />)
			.toJSON()
		expect(passwordInput).toMatchSnapshot()
	})
})