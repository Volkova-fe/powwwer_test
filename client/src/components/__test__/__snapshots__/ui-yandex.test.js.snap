// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тестирование компонента Login EmailInput без текста рендерится без ошибок 1`] = `
<div
  className="input__container "
>
  <div
    className="input pr-6 pl-6 input_type_email input_size_default"
    onClick={[Function]}
  >
    <label
      className="input__placeholder text noselect text_type_main-default"
    >
      E-mail
    </label>
    <input
      className="text input__textfield text_type_main-default"
      disabled={false}
      onBlur={[Function]}
      onFocus={[Function]}
      type="email"
    />
  </div>
</div>
`;

exports[`Тестирование компонента Login PasswordInput без текста рендерится без ошибок 1`] = `
<div
  className="input__container "
>
  <div
    className="input pr-6 pl-6 input_type_password input_size_default"
    onClick={[Function]}
  >
    <label
      className="input__placeholder text noselect text_type_main-default"
    >
      Пароль
    </label>
    <input
      className="text input__textfield text_type_main-default"
      disabled={false}
      onBlur={[Function]}
      onFocus={[Function]}
      type="password"
    />
    <div
      className="input__icon input__icon-action"
      onClick={[Function]}
    >
      <svg
        fill="#F2F2F3"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 4C14.0683 4 16.0293 4.71758 17.7366 6.04606C19.4439 7.36485 20.8976 9.29455 21.9415 11.7091C22.0195 11.8933 22.0195 12.1067 21.9415 12.2812C19.8537 17.1103 16.1366 20 12 20H11.9902C7.86341 20 4.14634 17.1103 2.05854 12.2812C1.98049 12.1067 1.98049 11.8933 2.05854 11.7091C4.14634 6.88 7.86341 4 11.9902 4H12ZM12 8.12121C9.8439 8.12121 8.09756 9.85697 8.09756 12C8.09756 14.1333 9.8439 15.8691 12 15.8691C14.1463 15.8691 15.8927 14.1333 15.8927 12C15.8927 9.85697 14.1463 8.12121 12 8.12121ZM12.0012 9.57362C13.3378 9.57362 14.4304 10.6597 14.4304 11.9979C14.4304 13.3264 13.3378 14.4124 12.0012 14.4124C10.6548 14.4124 9.56215 13.3264 9.56215 11.9979C9.56215 11.833 9.58166 11.6779 9.61093 11.5227H9.65971C10.7426 11.5227 11.6207 10.6694 11.6597 9.60272C11.767 9.58332 11.8841 9.57362 12.0012 9.57362Z"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`Тестирование компонента Login Кнопка c типом secondary рендерится без ошибок 1`] = `
<button
  className="button button_type_secondary button_size_medium"
/>
`;

exports[`Тестирование компонента Login Кнопка size="large" рендерится без ошибок 1`] = `
<button
  className="button button_type_primary button_size_large"
/>
`;

exports[`Тестирование компонента Login Кнопка size="medium" рендерится без ошибок 1`] = `
<button
  className="button button_type_primary button_size_medium"
/>
`;

exports[`Тестирование компонента Login Кнопка size="small" рендерится без ошибок 1`] = `
<button
  className="button button_type_primary button_size_small"
/>
`;

exports[`Тестирование компонента Login Кнопка без текста рендерится без ошибок 1`] = `
<button
  className="button button_type_primary button_size_medium"
/>
`;

exports[`Тестирование компонента Login Кнопка без текста рендерится без ошибок 2`] = `
<button
  className="button button_type_primary button_size_medium"
/>
`;

exports[`Тестирование компонента Login Кнопка с htmlType="button" рендерится без ошибок 1`] = `
<button
  className="button button_type_primary button_size_medium"
  type="button"
/>
`;

exports[`Тестирование компонента Login Кнопка с htmlType="submit" рендерится без ошибок 1`] = `
<button
  className="button button_type_primary button_size_medium"
  type="submit"
/>
`;
